pkg/inventory/consolelink.go

```diff
@@ -0,0 +1,65 @@
+package inventory
+
+import (
+	osconsolev1 "github.com/openshift/api/console/v1"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+// ConsoleLink represents the inventory of console links based on the current and desired states
+type ConsoleLink struct {
+	Create []osconsolev1.ConsoleLink
+	Update []osconsolev1.ConsoleLink
+	Delete []osconsolev1.ConsoleLink
+}
+
+// ForConsoleLinks builds an inventory of console links based on the existing and desired states
+func ForConsoleLinks(existing []osconsolev1.ConsoleLink, desired []osconsolev1.ConsoleLink) ConsoleLink {
+	update := []osconsolev1.ConsoleLink{}
+	mcreate := consoleLinkMap(desired)
+	mdelete := consoleLinkMap(existing)
+
+	for k, v := range mcreate {
+		if t, ok := mdelete[k]; ok {
+			tp := t.DeepCopy()
+			util.InitObjectMeta(tp)
+
+			// we can't blindly DeepCopyInto, so, we select what we bring from the new to the old object
+			tp.Spec = v.Spec
```

Comment:
This makes me wonder whether your `updateHref` is actually needed. This should be taking care of updating the Href already.

Reply:
May be I chose a bad name for the function, the propose of `updateHref` is to fetch the route, take the Host and assign it to the Href part of the console CR. The think is, the Host in the route CR is empty at the moment of creating the route.We need to create the route first, then get it to know the resulting host.